### ISSUE TITLE
Error via postfix: SASL CRAM-MD5 authentication failed

### ISSUE THREAD

**Issue by blueyed** [corecode/dma#22]
After upgrading my postfix server, I see the following error when trying to send a mail via `mail` + `dma`:

```
Sep 21 00:10:31 elfe postfix/smtpd[13234]: connect from unknown[external-ip]
Sep 21 00:10:31 elfe dovecot: auth-worker(13617): mysql(internal-ip): Connected to database foo
Sep 21 00:10:33 elfe postfix/smtpd[13234]: warning: unknown[external-ip]: SASL CRAM-MD5 authentication failed: xxx==
Sep 21 00:10:33 elfe postfix/smtpd[13234]: NOQUEUE: reject: RCPT from unknown[external-ip]: 554 5.7.1 <unknown[external-ip]>: Client host rejected: Access denied; from=<foo> to=<bar> proto=ESMTP helo=<baz>
Sep 21 00:10:33 elfe postfix/smtpd[13234]: lost connection after RCPT from unknown[external-ip]
Sep 21 00:10:33 elfe postfix/smtpd[13234]: disconnect from unknown[external-ip]
```

Local mail.log:

```
Sep 21 00:10:31 localhost dma[10048e]: new mail from user=user uid=1000 envelope_from=<user@localhost>
Sep 21 00:10:31 localhost dma[10048e]: mail to=<user@host> queued as 10048e.f238f0
Sep 21 00:10:31 localhost dma[10048e.f238f0]: trying delivery
Sep 21 00:10:31 localhost dma[10048e.f238f0]: using smarthost (host:587)
Sep 21 00:10:31 localhost dma[10048e.f238f0]: trying remote delivery to host [host-ip] pref 0
Sep 21 00:10:31 localhost dma[10048e.f238f0]: SSL initialization successful
Sep 21 00:10:31 localhost dma[10048e.f238f0]: using SMTP authentication for user user@host
Sep 21 00:10:33 localhost dma[10048e.f238f0]: remote delivery deferred: AUTH cram-md5 failed: 535 5.7.8 Error: authentication failed: xxx
Sep 21 00:10:33 localhost dma[10048e.f238f0]: SMTP login not available. Trying without.
Sep 21 00:10:33 localhost dma[10048e.f238f0]: remote delivery to host [host-ip] failed after RCPT TO: 554 5.7.1 <unknown[91.64.37.19]>: Client host rejected:
Sep 21 00:10:33 localhost dma[10048e.f238f0]: delivery failed, bouncing as 100b9a
Sep 21 00:10:33 localhost dma[100b9a]: new mail from user=user uid=1000 envelope_from=<>
Sep 21 00:10:33 localhost dma[100b9a]: mail to=<user> queued as 100b9a.f5b450
Sep 21 00:10:33 localhost dma[100b9a.f5b450]: trying delivery
Sep 21 00:10:33 localhost dma[100b9a.f5b450]: delivery successful
```

Delivery via Thunderbird and STARTTLS on the same port works.

I am using dma 0.9-1 from Debian/Ubuntu.

**Comment by cweiske** [comment]
Here (ubuntu 16.04, dma 0.9-1) it also shows the md5 error, but then it falls back to login and it works:

> Dec 20 16:00:22 mogiedyne dma[1120a5a.1093f60]: smarthost authentication: AUTH cram-md5 not available: 535 5.7.8 Error: authentication failed: no mechanism available
> Dec 20 16:00:23 mogiedyne dma[1120a5a.1093f60]: delivery successful

So all fine here.

**Comment by blueyed** [comment]
Closing as duplicate of #15.
